### PR TITLE
Initial `go test` for UC `tanzu` package

### PR DESCRIPTION
**Pull request by davidvonthenen**
## What this PR does / why we need it

Just like the title says.... It's just an initial effort to get something on the board. This can be refactored over time. I didn't include `go test` funcs for things that required a Unmanaged Cluster be running/available.

Results:
```
PASS
coverage: 20.9% of statements
ok  	github.com/vmware-tanzu/community-edition/cli/cmd/plugin/unmanaged-cluster/tanzu	2.567s
```

## Which issue(s) this PR fixes

This helps address the `go test` effort for the `tanzu` package. Issue found here: https://github.com/vmware-tanzu/community-edition/pull/4242

## Describe testing done for PR

`go test` passes

## Special notes for your reviewer

NA

### CODE REVIEW

**Review by jpmcb — DISMISSED** [review]
Nice!!!! 👏🏼

**Review by stmcginnis — DISMISSED** [review]
Looks great! Pointed out a couple minor things, but no worries if this merges with them.

File: cli/cmd/plugin/unmanaged-cluster/tanzu/tanzu_test.go [hunk 1]
```diff
@@ -0,0 +1,360 @@
+// Copyright 2021 VMware Tanzu Community Edition contributors. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply cli/cmd/plugin/unmanaged-cluster/tanzu/tanzu_test.go]: updated!

File: cli/cmd/plugin/unmanaged-cluster/tanzu/tanzu_test.go [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+		t.Error("compatibility path should always succeed")
+	}
+	if compatPath == "" {
+		t.Error("compatibility path should always return a valid compatPath")
+	}
+	if !strings.Contains(compatPath, homeDir) {
+		t.Errorf("compatibility path should always contain the homedir, was %s", compatPath)
+	}
+	if !strings.Contains(compatPath, ".config") {
+		t.Errorf("gcompatibility path should always contain .config in the the path, was %s", compatPath)
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply cli/cmd/plugin/unmanaged-cluster/tanzu/tanzu_test.go]: fixed!

**Review by stmcginnis — APPROVED** [review]
Nice!

**Review by jpmcb — APPROVED** [review]

**Pull request merged**